clang/lib/Parse/ParseDecl.cpp

```diff
@@ -488,6 +488,13 @@ unsigned Parser::ParseAttributeArgsCommon(
   bool AttributeHasVariadicIdentifierArg =
       attributeHasVariadicIdentifierArg(*AttrName, Form.getSyntax(), ScopeName);
 
+  if (Tok.is(tok::hash) || Tok.is(tok::hashhash)) {
```

Comment:
@cor3ntin do we need to distinguish between valid preprocessor tokens and invalid post-phase-7 tokens?

I *think* the only tokens that aren't valid post phase 7 is hash, but I'm not certain what we're supposed to do with things like `%:` `%:%:` which are digraphs for hash. Are those supposed to also be rejected?

Reply:
We are supposed to reject `%:`, yes
https://eel.is/c++draft/lex.operators#nt:operator-or-punctuator